core/render2d/src/render_internal.cpp

```diff
@@ -867,90 +881,93 @@ void MoleculeRenderInternal::_cloneAndFillMappings()
     _own_mol = true;
 }
 
-void MoleculeRenderInternal::_prepareSGroups()
+void MoleculeRenderInternal::_prepareSGroups(bool collapseAtLeastOneSuperatom)
 {
     _cloneAndFillMappings();
 
     BaseMolecule& mol = *_mol;
-    if (_opt.collapseSuperatoms)
+    if (collapseAtLeastOneSuperatom)
     {
         for (int i = mol.sgroups.begin(); i != mol.sgroups.end(); i = mol.sgroups.next(i))
         {
             SGroup& sgroup = mol.sgroups.getSGroup(i);
-            if (sgroup.sgroup_type == SGroup::SG_TYPE_SUP)
+            if (abs(sgroup.contracted) == 1)
             {
-                const Superatom& group = (Superatom&)sgroup;
-                Vec3f centre;
-                for (int i = 0; i < group.atoms.size(); ++i)
+                if (sgroup.sgroup_type == SGroup::SG_TYPE_SUP)
                 {
-                    int aid = group.atoms[i];
-                    centre.add(mol.getAtomXyz(aid));
-                }
-                centre.scale(1.0f / group.atoms.size());
-                int said = -1;
+                    const Superatom& group = (Superatom&)sgroup;
+                    Vec3f centre;
+                    for (int i = 0; i < group.atoms.size(); ++i)
+                    {
+                        int aid = group.atoms[i];
+                        centre.add(mol.getAtomXyz(aid));
+                    }
+                    centre.scale(1.0f / group.atoms.size());
+                    int said = -1;
```

Comment:
rename variable to user friendly

Reply:
The "said" variable was introduced earlier without any comments. I have just rearanged the code. From the code, I assume that "said" stands for Start Atom ID of the bond. There is also "naid" that could mean eNd Atom ID of the bond, and there is "aid" which could mean just Aton ID as it gets index of an atom.
If there are no objections against this assumption I propose to name them as aid : atomID; naid : endAtomID; said : startAtomID